src/els_xref_diagnostics.erl

```diff
@@ -62,6 +62,12 @@ make_diagnostic(#{range := Range, id := Id}) ->
                                  , source()).
 
 -spec has_definition(poi(), els_dt_document:item()) -> boolean().
+has_definition(#{ kind := application
+                , id := {erlang, module_info, 0} }, _) -> true;
+has_definition(#{ kind := application
+                , id := {erlang, module_info, 1} }, _) -> true;
```

Comment:
This seems wrong. Those functions are not defined on the erlang module. Why would they show up like this in here?

Reply:
Empirical testing, I logged what I got when I opened files with the issue.

And it *does* work with this match.

Reply:
I wonder if it's related to this https://github.com/erlang-ls/erlang_ls/issues/633 - since those functions also exist on the `erlang` module

Reply:
The `module_info/0,1` functions are special and inserted by the compiler to **all** modules, not just the `erlang` one:

https://erlang.org/doc/reference_manual/modules.html#module_info-0-and-module_info-1-functions

I suggest we hold on for this one and instead focus on #621 , which will make this not needed.